src/qibolab/instruments/qm/driver.py

```diff
@@ -103,7 +103,7 @@ def fetch_results(result, ro_pulses):
         results = {}
         for qmpulse in ro_pulses:
             pulse = qmpulse.pulse
-            results[pulse.qubit] = results[pulse.serial] = qmpulse.acquisition.fetch(
```

Comment:
@stavros11 here I'm left with a single error
```py
                for op in self._config["elements"][qe]["operations"]:
>                   if op.find("baked") != -1:
E                   AttributeError: 'int' object has no attribute 'find'
                                                                                                                                                                                      
.venv/lib/python3.11/site-packages/qualang_tools/bakery/bakery.py:116: AttributeError
```

It seems like QM wants to check whether the pulse is baked or not based on the key. Currently, I'm failing just because I'm using an `int` as key, while they expect a `str`.

I will quickfix this by stringifying the `int`, but I wonder if I'm doing something wrong, because the former `.serial` actually contained the substring `"bakend"` somewhere... (on purpose, of course)

Reply:
Actually, it was required in multiple places. Check 8c97dd7 for the relevant commit.

Reply:
Sorry, I just saw this. Note that in #738 I stopped using the serial, I replaced it with a string that looks like the serial but without the `start` xD. But this should solve this problem (will have conflicts though).

Before I wasn’t using serial really by accident, I just needed to register the pulse using a string in the QM config and serial was a candidate. Any string works, but if it’s also kind of readable debugging is easier.

Anyway, I understand this is resolved now?

Reply:
Yes, it is solved.

I know sometimes debugging might get harder because of some choices, but I'd rather improve the *infrastructure* for debugging separately from the main package itself (though this is debatable).

However, `"baked"` will never happen to be part of the string, but it should have been the case even before (no reason why `.serial` shoud have included `"baked"`, unless it was part of the name of a qubit...).